truffle/src/main/java/org/truffleruby/parser/BodyTranslator.java

```diff
@@ -1729,7 +1729,7 @@ public RubyNode visitGlobalAsgnNode(GlobalAsgnParseNode node) {
             rhs = new CheckOutputSeparatorVariableTypeNode(rhs);
             rhs.unsafeSetSourceSection(sourceSection);
         } else if (name.equals("$_")) {
-            if (getSourcePath(sourceSection).endsWith(buildPartialPath("truffle", "rubysl", "rubysl-stringio", "lib", "rubysl", "stringio", "stringio.rb"))) {
+            if (getSourcePath(sourceSection).endsWith(buildPartialPath("rubysl", "rubysl-stringio", "lib", "rubysl", "stringio", "stringio.rb"))) {
```

Comment:
Also unrelated, but these sorts of string comparisons for path operations is the biggest reason we don't run on Windows. I realize it's not a going concern, but using the `java.nio.file.Path` API neatly avoids that.